### PR TITLE
PgSqlDriver: fixed formatLike() thx @milo [Closes #46]

### LINKED ISSUE #46: Postgre: buggy formatLike()
See https://github.com/dg/dibi/pull/160

### PR DESCRIPTION
**Pull request by dg**
[Closes #46]

### CODE REVIEW

**Comment by milo** [comment]
The test is failing for me, but I guess it is false positive. Dibi tests pass. Strange, I'll check it.

**Comment by milo** [comment]
@dg Dibi [escapes](https://github.com/dg/dibi/blob/020b15c0e29fd14d85dd510d560795d6d409cb6b/dibi/libs/DibiTranslator.php#L581) (security I guess) inlined SQL quoted tokens. So the test passes for dibi.

File: src/Database/Drivers/PgSqlDriver.php [hunk 1]
```diff
@@ -87,7 +87,8 @@ public function formatDateTime(/*\DateTimeInterface*/ $value)
 	 */
 	public function formatLike($value, $pos)
 	{
-		$value = strtr($value, array("'" => "''", '\\' => '\\\\', '%' => '\\\\%', '_' => '\\\\_'));
+		$bs = substr($this->connection->quote('\\', \PDO::PARAM_STR), 1, -1); // standard_conforming_strings = on/off
```
Comment (milo) [inline src/Database/Drivers/PgSqlDriver.php]: Value escaping is missing. There must be:
```php
$value = substr($this->connection->quote($value, \PDO::PARAM_STR), 1, -1);
```

File: tests/Database/PgSqlDriver.formatLike.phpt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	Assert::false($connection->query("SELECT 'AAxBB' LIKE", $connection::literal($driver->formatLike('A_B', 0)))->fetchField());
+	Assert::true( $connection->query("SELECT 'AA_BB' LIKE", $connection::literal($driver->formatLike('A_B', 0)))->fetchField());
+
+	Assert::false($connection->query("SELECT 'AAxBB' LIKE", $connection::literal($driver->formatLike('A%B', 0)))->fetchField());
+	Assert::true( $connection->query("SELECT 'AA%BB' LIKE", $connection::literal($driver->formatLike('A%B', 0)))->fetchField());
+
+	Assert::same('AA\\BB', $connection->query("SELECT 'AA\\BB'")->fetchField());
+	Assert::false($connection->query("SELECT 'AAxBB' LIKE", $connection::literal($driver->formatLike('A\\B', 0)))->fetchField());
+	Assert::true( $connection->query("SELECT 'AA\\BB' LIKE", $connection::literal($driver->formatLike('A\\B', 0)))->fetchField());
+};
```
Comment (milo) [inline tests/Database/PgSqlDriver.formatLike.phpt]: What about this tests?
```php
	$driver = $connection->getSupplementalDriver();
	Assert::false($connection->query("SELECT 'AAxBB' LIKE", $connection::literal($driver->formatLike('A_B', 0)))->fetchField());
	Assert::true( $connection->query("SELECT 'AA_BB' LIKE", $connection::literal($driver->formatLike('A_B', 0)))->fetchField());

	Assert::false($connection->query("SELECT 'AAxBB' LIKE", $connection::literal($driver->formatLike('A%B', 0)))->fetchField());
	Assert::true( $connection->query("SELECT 'AA%BB' LIKE", $connection::literal($driver->formatLike('A%B', 0)))->fetchField());

	Assert::false($connection->query("SELECT ? LIKE", 'AAxBB', $connection::literal($driver->formatLike('A\\B', 0)))->fetchField());
	Assert::true( $connection->query("SELECT ? LIKE", 'AA\\BB', $connection::literal($driver->formatLike('A\\B', 0)))->fetchField());

	Assert::false($connection->query("SELECT ? LIKE", 'AAxBB', $connection::literal($driver->formatLike("A'B", 0)))->fetchField());
	Assert::true( $connection->query("SELECT ? LIKE", "AA'BB", $connection::literal($driver->formatLike("A'B", 0)))->fetchField());

	Assert::false($connection->query("SELECT ? LIKE", 'AAxBB', $connection::literal($driver->formatLike('A"B', 0)))->fetchField());
	Assert::true( $connection->query("SELECT ? LIKE", 'AA"BB', $connection::literal($driver->formatLike('A"B', 0)))->fetchField());
```

The quotes escaping test is covered by [another test](https://github.com/dg/dibi/blob/020b15c0e29fd14d85dd510d560795d6d409cb6b/tests/dibi/DibiTranslator.phpt#L205) in dibi.

Comment (dg) [inline tests/Database/PgSqlDriver.formatLike.phpt]: @milo Nechtěl by jsi poslat ten PR? Tuhle DB jsem nikdy nepoužíval a fakt nevím která bije…

**Pull request closed without merging**